src/credentials/FabricTable.cpp

```diff
@@ -973,10 +1032,42 @@ CHIP_ERROR FabricTable::Init(PersistentStorageDelegate * storage)
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR FabricTable::Init(PersistentStorageDelegate * storage, OperationalKeystore * operationalKeystore)
+void FabricTable::Forget(FabricIndex fabricIndex)
 {
-    mOperationalKeystore = operationalKeystore;
-    return Init(storage);
+    VerifyOrReturn(IsValidFabricIndex(fabricIndex));
+
+    ChipLogProgress(FabricProvisioning, "Forgetting fabric 0x%x", static_cast<unsigned>(fabricIndex));
+
+    auto * fabricInfo = FindFabricWithIndex(fabricIndex);
+    VerifyOrReturn(fabricInfo != nullptr);
+
+    RevertPendingFabricData();
+    fabricInfo->Reset();
```

Comment:
This would seem to change the in-memory view of the Fabric Table to be different than the persisted view.  Should we actually allow this?

This seems odd because it's asymmetric with respect to init vs. shutdown.  When we init the Fabric Table, it loads all fabrics.  But now we are offering an interface to selectively remove Fabric info out of memory.  As called from the controller code, I think this means that once we shutdown a fabric-scoped controller, we can't reinitialize it again until we reinitialize the entire fabric table.

Reply:
OK, that's fine.  Other implementations that use the FabricTable don't need to call Forget if they don't want to.  That's good enough for me.

Thanks for the background.